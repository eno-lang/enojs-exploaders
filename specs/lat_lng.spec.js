const eno = require('enojs');
const loaders = require('../index.js');

describe('latLng', () => {

  let document;

  describe('48.205870, 16.413690', () => {
    it('returns a lat/lng object', () => {
      document = eno.parse('coordinates: 48.205870, 16.413690');
      expect(document.field('coordinates', loaders.latLng)).toEqual({ lat: 48.205870, lng: 16.413690 });
    });
  });

  describe('48.205870,', () => {
    it('throws an error', () => {
      document = eno.parse('coordinates: 48.205870,');
      expect(() => document.field('coordinates', loaders.latLng)).toThrowErrorMatchingSnapshot();
    });
  });

  describe(', 16.413690', () => {
    it('throws an error', () => {
      document = eno.parse('coordinates: , 16.413690');
      expect(() => document.field('coordinates', loaders.latLng)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('48,205870, 16,413690', () => {
    it('throws an error', () => {
      document = eno.parse('coordinates: 48,205870, 16,413690');
      expect(() => document.field('coordinates', loaders.latLng)).toThrowErrorMatchingSnapshot();
    });
  });
});
