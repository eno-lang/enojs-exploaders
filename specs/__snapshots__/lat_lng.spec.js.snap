// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`latLng , 16.413690 throws an error 1`] = `
"coordinates must contain lat/lng coordinates formatted xx.xxxxxx, xx.xxxxxx

   Line | Content
 >    1 | coordinates: , 16.413690
"
`;

exports[`latLng 48,205870, 16,413690 throws an error 1`] = `
"coordinates must contain lat/lng coordinates formatted xx.xxxxxx, xx.xxxxxx

   Line | Content
 >    1 | coordinates: 48,205870, 16,413690
"
`;

exports[`latLng 48.205870, throws an error 1`] = `
"coordinates must contain lat/lng coordinates formatted xx.xxxxxx, xx.xxxxxx

   Line | Content
 >    1 | coordinates: 48.205870,
"
`;
